mobile/lib/modules/memories/providers/memory.provider.dart

```diff
@@ -2,9 +2,26 @@ import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:immich_mobile/modules/memories/models/memory.dart';
 import 'package:immich_mobile/modules/memories/services/memory.service.dart';
 
+/// A notifier to be used with the Memory Lane service. Sets the date, and
+/// will update the date to now on app resume so that the memory provider
+/// can update accordingly
+class TodayNotifier extends StateNotifier<DateTime> {
+  TodayNotifier() : super(DateTime.now());
+
+  /// Sets the time to now to refresh today
+  void update() {
+    state = DateTime.now();
+  }
+}
+
+final todayProvider = StateNotifierProvider<TodayNotifier, DateTime>((ref) {
+  return TodayNotifier();
+});
+
 final memoryFutureProvider =
     FutureProvider.autoDispose<List<Memory>?>((ref) async {
   final service = ref.watch(memoryServiceProvider);
+  final today = ref.watch(todayProvider);
```

Comment:
I kind of want to update this `todayProvider` right here. Because this will not update automatically until an app resume... 

Maybe I should just check `DateTime.now()` here? I don't like updating the state of a provider when accessing another provider.

Reply:
In other words, this patch will introduce the following bug:

- Open Immich at 11:59PM.
- Navigate to Search at 12:00AM.
- Navigate back to Photos at 12:01AM
- Memory lane photos will not update

This is because the `todayProvider` will only update its state on app resume. Something else should really be updating its state, but I don't know who without introducing this side effect in this provider.